verticapy/pipeline/_train.py

```diff
@@ -0,0 +1,224 @@
+        ``list`` of the columns needed to deploy the model.
+
+    Returns
+    -------
+    return meta_sql, model, model_sql
+    str
+        The SQL to replicate the steps of the yaml file.
+    VerticaModel
+        The model created after training.
+    str
+        The SQL needed to retrain the model.
+    """
+    meta_sql = ""
+    if "train_test_split" in train:
+        info = train["train_test_split"]
+        test_size = 0.33 if "test_size" not in info else info["test_size"]
+        train_set, test_set = vdf.train_test_split(test_size)
+        meta_sql += execute_and_return(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TRAIN_VIEW'} AS SELECT * FROM "
+            + train_set.current_relation()
+            + ";"
+        )
+        meta_sql += execute_and_return(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TEST_VIEW'} AS SELECT * FROM "
+            + test_set.current_relation()
+            + ";"
+        )
+
+    else:
+        meta_sql += execute_and_return(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TRAIN_VIEW'} AS SELECT * FROM "
+            + vdf.current_relation()
+            + ";"
+        )
+        meta_sql += execute_and_return(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TEST_VIEW'} AS SELECT * FROM "
+            + vdf.current_relation()
+            + ";"
+        )
+
+    methods = list(train.keys())
+    methods = [method for method in methods if "method" in method]
+    for method in methods:
+        tm = train[method]
+        name = tm["name"]
+        target = tm["target"] if "target" in tm else ""
+        params = tm["params"] if "params" in tm else []
+        param_str = ""
+        for param in params:
+            temp = params[param]
+            if isinstance(temp, str):
+                param_str += f"{param} = '{params[param]}', "
+            else:
+                param_str += f"{param} = {params[param]}, "
+
+        param_str = param_str[:-2]
+        model = eval(
+            f"{name}('{pipeline_name + '_MODEL'}', {param_str})",
+            globals(),
+        )
+        predictors = cols  # ['"col1"', '"col2"', '"col3"', '"col4"']
+        if "include" in tm:
+            predictors = tm["include"]
+
+        if "exclude" in tm:
+            predictors = list(set(predictors) - set(tm["exclude"]))
+
+        if target == "":
+            # UNSUPERVISED
+            model.fit(pipeline_name + "_TRAIN_VIEW", predictors)
+        else:
+            # SUPERVISED
+            model.fit(pipeline_name + "_TRAIN_VIEW", predictors, target)
+        meta_sql += "\n"
+
+        ts = model.get_vertica_attributes()  # TableSample
+        if "call_string" in ts["attr_name"]:
+            # SUPERVISED
+            model_sql = model.get_vertica_attributes("call_string")["call_string"][0]
+            if model_sql.split(" ")[0] != "SELECT":
+                model_sql = "SELECT " + model_sql + ";"
+        else:
+            # UNSUPERVISED
```

Comment:
Do unsupervised algorithms really have no call string?

Reply:
No it seems sporadically there is a few algorithms that don't have a call string. In my initial tests I saw LinearSVC had it and KMeans did not, so I generalized to the two types (supervised and unsupervised).